examples/arithmetics/src/language-server/arithmetics.langium

```diff
@@ -26,7 +26,14 @@ Addition infers Expression:
     Multiplication ({infer BinaryExpression.left=current} operator=('+' | '-') right=Multiplication)*;
 
 Multiplication infers Expression:
-    PrimaryExpression ({infer BinaryExpression.left=current} operator=('*' | '/') right=PrimaryExpression)*;
+    Potentiation ({infer BinaryExpression.left=current} operator=('*' | '/') right=Potentiation)*;
+
+Potentiation infers Expression:
```

Comment:
The potentiation operator is right-associative. Your rule will make it left-associative. It is just a good old pattern for compiler creators.
You can see the right-associative pattern in this article [here](https://www.typefox.io/blog/parsing-expressions-with-xtext).
Should be something like:
```
Potentiation infers Expression:
 PrimaryExpression ({BinaryExpression.left=current} '^' right=Potentiation)?;
```

The syntax tree will then grow to the right and not the left. You can draw the syntax tree for practicing. 4^3^2 will have two parse trees, but only one is the commonly used one.